### PR TITLE
[#950] Add `pStateIsInternallyConsistent` property

### LINKED ISSUE #950: Check PState consistency across a POOL trace
Add a property to check that the various maps of a PState stay in sync as we step through the signals of a POOL trace.

### PR DESCRIPTION
**Pull request by mhuesch**
Also creates a new test group for POOL and relocates the RegPool
property to it.

Closes #950.

### CODE REVIEW

**Comment by mhuesch** [comment]
I'm signing off soon, and won't be back online until after my leave. Can someone take care of this PR - adding changes if needed and merging it? Or, since it's a somewhat independent property, the branch probably won't get too stale.

File: shelley/chain-and-ledger/executable-spec/test/Rules/TestPool.hs [hunk 1]
```diff
@@ -136,3 +139,30 @@ registeredPoolIsAdded env ssts =
...
+    let StakePools stPoolsMap = stPools_
+        poolKeys = M.keysSet stPoolsMap
+        pParamKeys = M.keysSet pParams_
+        retiringKeys = M.keys retiring_
+        cCountersKeys = M.keysSet cCounters_
+
+    sequence_ [ -- These 3 key sets should be equal.
+                poolKeys === pParamKeys
+              , pParamKeys === cCountersKeys
+              , cCountersKeys === poolKeys
```
Comment (mgudemann) [inline shelley/chain-and-ledger/executable-spec/test/Rules/TestPool.hs]: the third check seems redundant

File: shelley/chain-and-ledger/executable-spec/test/Rules/TestPool.hs [hunk 2]
```diff
@@ -136,3 +139,30 @@ registeredPoolIsAdded env ssts =
...
+                -- A retiring pool should still be registered in `stPools`.
+              , traverse_ (assert . (`S.member` poolKeys)) retiringKeys
+              ]
+   where
```
Comment (mgudemann) [inline shelley/chain-and-ledger/executable-spec/test/Rules/TestPool.hs]: redundant `where`

**Comment by mgudemann** [comment]
I am happy to take care of this PR this week

**Pull request merged**